mesonbuild/linkers/linkers.py

```diff
@@ -700,14 +700,31 @@ def build_rpath_args(self, env: 'Environment', build_dir: str, from_dir: str,
         return (args, rpath_dirs_to_remove)
 
     def get_win_subsystem_args(self, value: str) -> T.List[str]:
-        if 'windows' in value:
-            args = ['--subsystem,windows']
-        elif 'console' in value:
-            args = ['--subsystem,console']
-        else:
-            raise MesonException(f'Only "windows" and "console" are supported for win_subsystem with MinGW, not "{value}".')
+        # Subsystem affects a single entry at the beginning of PE executables.
+        # It is used to verify that an executable is of the expected
+        # application type before attempting to execute it.
+        subsystems = {
+            "native": "1",
+            "windows": "2",
+            "console": "3",
+            "posix": "7",
+            "efi_application": "10",
+            "efi_boot_service_driver": "11",
+            "efi_runtime_driver": "12",
+            "efi_rom": "13",
+            "xbox": "14",
```

Comment:
xbox is not a valid value in https://github.com/mesonbuild/meson/blob/7650f328284f24c2f70dcfb1689ae8fc22c04cf1/mesonbuild/build.py#L1260-L1264